### PR TITLE
Add Docker compose for local db & emailSender

### PR DESCRIPTION
**Pull request by adrienmntr**
Set up the local db with docker-compose
Send email for password recovery

### CODE REVIEW

File: .python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+sso
```
Comment (Piwakk) [inline .python-version:1]: Please remove the file

File: sso/.env.dist [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
 # Should begin with `-----BEGIN PUBLIC KEY-----` and end with `-----END PUBLIC KEY-----`.
 Please pay attention to the new lines. They should be encoded as `\\n` in this file.
-PRIVATE_KEY=""
-PUBLIC_KEY=""
+PRIVATE_KEY=
+PUBLIC_KEY=
 
-REDIRECT_URLS=key1=value1,key2=value2
\ No newline at end of file
+REDIRECT_URLS=
```
Comment (Piwakk) [inline sso/.env.dist:23]:
```suggestion
# Syntax: `key1=value1,key2=value2`
REDIRECT_URLS=
```

File: sso_server/email_template/passwordRecovery.html [hunk 3]
```diff
@@ -0,0 +1,160 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml" xmlns:o="urn:schemas-microsoft-com:office:office" style="width:100%;font-family:helvetica, 'helvetica neue', arial, verdana, sans-serif;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%;padding:0;Margin:0">
+ <head>
+  <meta charset="UTF-8">
+  <meta content="width=device-width, initial-scale=1" name="viewport">
+  <meta name="x-apple-disable-message-reformatting">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta content="telephone=no" name="format-detection">
+  <title>Nouveau modèle de courrier électronique 2020-07-17</title>
```
Comment (Piwakk) [inline sso_server/email_template/passwordRecovery.html:9]: ?

File: sso_server/mail.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    def connect(self):
+        self.smtp = smtplib.SMTP_SSL(settings.EMAIL["HOST"], settings.EMAIL["PORT"])
+        self.smtp.ehlo()
+        pw = settings.EMAIL["PASSWORD"]
+        self.smtp.login(self.account, pw)
+
+    def sendemail(self, to, subject, text, cc=None, bcc=None, attachment=None, attachments=None, template=False, template_file_path=None, **kwargs):
+        msg = MIMEMultipart('alternative')
+        msg['Subject'] = "[Rezal] {}".format(subject)
+        msg['From'] = '{} <{}>'.format(self.name, self.account)
```
Comment (Piwakk) [inline sso_server/mail.py:26]: Use f-strings, it's simpler:

```suggestion
        msg['Subject'] = f"[Rezal] {subject}"
        msg['From'] = f'{self.name} <{self.account}>'
```

File: sso_server/mail.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        else:
+            content = text
+
+        msg.attach(MIMEText(content, 'html'))
+        if attachment is not None:
+            part = MIMEApplication(
+                attachment[0].read(),
+                Name=attachment[1]
+            )
+            part['Content-Disposition'] = 'attachment; filename="{}"'.format(attachment[1])
```
Comment (Piwakk) [inline sso_server/mail.py:49]: Use f-strings

File: sso_server/mail.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            msg.attach(part)
+        if attachments is not None:
+            for attachment in attachments:
+                part = MIMEApplication(
+                    attachment[0].read(),
+                    Name=attachment[1])
+                part['Content-Disposition'] = 'attachment; filename="{}"'.format(attachment[1])
```
Comment (Piwakk) [inline sso_server/mail.py:56]: Use f-strings

File: sso_server/mail.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                msg.attach(part)
+        if len(to) > 0:
+            self.smtp.sendmail(self.account, destinations, msg.as_string())
+
+    def send_passwordrecovery_link(self, user_address, name, passwordrecovery_id):
+        link = settings.FRONTEND_HOST + "/mot-de-passe/nouveau/" + passwordrecovery_id
+        text = "Cliquez sur ce lien pour reéinitialiser votre mot de passe : " + link
```
Comment (Piwakk) [inline sso_server/mail.py:63]:
```suggestion
        text = "Cliquez sur ce lien pour réinitialiser votre mot de passe : " + link
```

File: sso_server/mail.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            for attachment in attachments:
+                part = MIMEApplication(
+                    attachment[0].read(),
+                    Name=attachment[1])
+                part['Content-Disposition'] = 'attachment; filename="{}"'.format(attachment[1])
+                msg.attach(part)
+        if len(to) > 0:
+            self.smtp.sendmail(self.account, destinations, msg.as_string())
+
+    def send_passwordrecovery_link(self, user_address, name, passwordrecovery_id):
```
Comment (Piwakk) [inline sso_server/mail.py:61]: Please follow the snake case convention: `send_password_recovery_link`

Comment (Piwakk) [reply sso_server/mail.py:61]: And `password_recovery_id`

File: sso_server/views.py [hunk 9]
```diff
@@ -151,15 +154,23 @@ def post(self, request, *args, **kwargs):
...
+            PasswordRecovery.objects.all().update(used=1)
+
         # Create a PasswordRecovery.
         serializer = PasswordRecoverySerializer(data={"user": user.email})
 
         if not serializer.is_valid():
             return self.INVALID_EMAIL_ERROR
 
-        serializer.save(user=user)
+        passwordrecovery = serializer.save(user=user)
```
Comment (Piwakk) [inline sso_server/views.py:166]:
```suggestion
        password_recovery = serializer.save(user=user)
```

File: sso_server/email_template/passwordRecovery.html (hunk 3 above)
Comment (adrienmntr) [reply sso_server/email_template/passwordRecovery.html:9]: Just the html template for the passwordrecovery email

Comment (Piwakk) [reply sso_server/email_template/passwordRecovery.html:9]: Sorry, I was hinting at the title which does not seem relevant

Comment (adrienmntr) [reply sso_server/email_template/passwordRecovery.html:9]: password_recovery_email_template.html ?

Comment (Piwakk) [reply sso_server/email_template/passwordRecovery.html]: `<title>Nouveau modèle de courrier électronique 2020-07-17</title>`

**Comment by Piwakk** [comment]
I see that `frontend/public/bootstrap.min.css` is included in your PR.
Could you remove it please? Thanks

**Comment by delmaass** [comment]
@Piwakk I just removed `frontend/public/bootstrap.min.css` :)

**Comment by abocquet** [comment]
There are a lot of diffs linked to reformatting, could we either
- cancel them if they are not black/prettier compatible
- run black and prettier in another PR and then see what truly changed here?

**Comment by delmaass** [comment]
There is a conflict between Blackend and Prettier CI, this PR fixes it #18

**Comment by abocquet** [comment]
Then could you create a PR called (fix #18 ) and make the changes there? "Add Docker compose for local db & emailSender" is not a very good clue that we are doing that here, and there are a number of changes that are not related to fixing the issue  😉

**Comment by delmaass** [comment]
I think there is a misunderstanding ahah
The PR #18 fixes the conflict between Blackend and Prettier CI and applies Black to the backend and Prettier to the frontend in order to pass the "new" CI workflow.
Once the PR #18 is merged, we could come back here to see what truly changed.

**Comment by abocquet** [comment]
Ok my bad, I misunderstood 🙇 
I merged #18 , there are a couple of merge conflicts (we could kill two birds with one stone by solving the merge conflict and #21 at once)

**Review by abocquet — APPROVED** [review]

**Pull request merged**